### PR TITLE
Update olm images to use openshift registry instead of quay

### PR DESCRIPTION
**Pull request by ecordell**
this should be the second to last update for OLM - this is updating the images (want to find any problems with that) and there are some changes coming through to the catalog that we'll follow up with

### CODE REVIEW

File: roles/olm/defaults/main.yaml [hunk 1]
```diff
@@ -1,3 +1,5 @@
 ---
 operator_lifecycle_manager_remove: false
 operator_lifecycle_manager_install: true
+olm_operator_image: "{{ l_osm_registry_url | regex_replace('${component}' | regex_escape, 'operator-lifecycle-manager') }}"
+olm_catalog_operator_image: "{{ l_osm_registry_url | regex_replace('${component}' | regex_escape, 'operator-lifecycle-manager') }}"
```
Comment (sdodson) [inline roles/olm/defaults/main.yaml]: Will these always be the same image? Looks like prior to this PR they were different, not sure if this is a copy/paste error or not.
quay.io/coreos/olm versus quay.io/coreos/catalog

Comment (sdodson) [reply roles/olm/defaults/main.yaml]: Also, l_osm_registry_url will be 'docker.io/openshift/origin-${component}:${version}' for origin. Will that work for origin images too?
If not then we need something like this where we just rely on the replacement being a noop.

```
olm_registry_url_dict:
  origin: 'quay.io/coreos/operator-livecycle-manager'
  openshift-enterprise: "{{ l_osm_registry_url }}"
olm_operator_image: "{{ olm_registry_url_dict[openshift_deployment_type] | regex_replace('${component}' | regex_escape, 'operator-lifecycle-manager') }}"
olm_catalog_operator_image: "{{ olm_registry_url_dict[openshift_deployment_type] | regex_replace('${component}' | regex_escape, 'operator-lifecycle-manager') }}"
```

Comment (ecordell) [reply roles/olm/defaults/main.yaml]: For simplicity for now, we just threw both binaries into one image. I'll update with the quay urls for origin, though

Comment (ecordell) [reply roles/olm/defaults/main.yaml]: @sdodson Updated the image vars.

Is there a standard practice for how/when to tag origin images?

Comment (sdodson) [reply roles/olm/defaults/main.yaml]: Not that I know, we should probably establish a norm. In the past we'd just assumed everything was in docker.io/openshift.

**Pull request merged**